profile_sample.go

```diff
@@ -0,0 +1,84 @@
+package sentry
+
+// Based on https://github.com/getsentry/vroom/blob/d11c26063e802d66b9a592c4010261746ca3dfa4/internal/sample/sample.go
+// * unused fields are commented out
+// * some types are replaced with their representation in sentry-go
+
+import (
+	"time"
+)
+
+type (
+	profileDevice struct {
+		Architecture   string `json:"architecture"`
+		Classification string `json:"classification"`
+		Locale         string `json:"locale"`
+		Manufacturer   string `json:"manufacturer"`
+		Model          string `json:"model"`
+	}
+
+	profileOS struct {
+		BuildNumber string `json:"build_number"`
+		Name        string `json:"name"`
+		Version     string `json:"version"`
+	}
+
+	profileRuntime struct {
+		Name    string `json:"name"`
+		Version string `json:"version"`
+	}
+
+	profileSample struct {
+		ElapsedSinceStartNS uint64       `json:"elapsed_since_start_ns"`
+		QueueAddress        string       `json:"queue_address,omitempty"`
+		StackID             int          `json:"stack_id"`
+		State               profileState `json:"state,omitempty"`
```

Comment:
This is more of a thread state rather than a profile state. Clients don't really report it, it's really meant for the backend to tag the stack as "idle" when we the stack is empty. Not sure if we'd see this in Go.

I would get rid of it for now.